backend/tracim_backend/applications/upload_permissions/email_manager.py

```diff
@@ -0,0 +1,242 @@
+from email.message import Message
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+from email.utils import formataddr
+import typing
+
+from tracim_backend.applications.upload_permissions.models_in_context import (
+    UploadPermissionInContext,
+)
+from tracim_backend.lib.core.workspace import WorkspaceApi
+from tracim_backend.lib.mail_notifier.notifier import EmailManager
+from tracim_backend.lib.mail_notifier.sender import EmailSender
+from tracim_backend.lib.mail_notifier.sender import send_email_through
+from tracim_backend.lib.utils.logger import logger
+from tracim_backend.lib.utils.translation import Translator
+from tracim_backend.models.auth import User
+from tracim_backend.models.context_models import ContentInContext
+from tracim_backend.models.data import Workspace
+
+
+class UploadPermissionEmailManager(EmailManager):
+    def notify_new_upload(
+        self,
+        uploader_username,
+        workspace: Workspace,
+        uploaded_contents: typing.List[ContentInContext],
+    ) -> None:
+        email_sender = EmailSender(
+            self.config, self._smtp_config, self.config.EMAIL__NOTIFICATION__ACTIVATED
+        )
+        notifiable_roles = WorkspaceApi(
+            current_user=None, session=self.session, config=self.config
+        ).get_notifiable_roles(workspace, force_notify=True)
+
+        for role in notifiable_roles:
+            logger.info(
+                self,
+                'Generating new upload notification in workspace "{}" from "{}" to "{}"'.format(
+                    workspace.workspace_id, uploader_username, role.user.email
+                ),
+            )
+            translator = Translator(app_config=self.config, default_lang=role.user.lang)
+
+            message = self._notify_new_upload(
+                workspace=workspace,
+                receiver=role.user,
+                uploader_username=uploader_username,
+                translator=translator,
+                uploaded_contents=uploaded_contents,
+            )
+            send_email_through(
+                config=self.config, sendmail_callable=email_sender.send_mail, message=message
+            )
+
+    def _notify_new_upload(
```

Comment:
Method here should name like "_create_notify_new_upload_message". Current name suggest notification is done.

Reply:
issue about this there https://github.com/tracim/tracim/issues/2206
Does concern also share app, so not fixed here.